src/command/verify/verify.c

```diff
@@ -994,33 +1505,68 @@ verifyProcess(unsigned int *errorTotal)
                         ProtocolParallelJob *job = protocolParallelResult(parallelExec);
                         unsigned int processId = protocolParallelJobProcessId(job);
                         StringList *filePathLst = strLstNewSplit(varStr(protocolParallelJobKey(job)), FSLASH_STR);
+
+                        // Remove the result and file type identifier and recreate the path file name
+                        const String *resultId = strLstGet(filePathLst, 0);
+                        strLstRemoveIdx(filePathLst, 0);
+                        const String *fileType = strLstGet(filePathLst, 0);
                         strLstRemoveIdx(filePathLst, 0);
                         String *filePathName = strLstJoin(filePathLst, "/");
 
+                        // Initialize the result sets
                         VerifyArchiveResult *archiveIdResult = NULL;
+                        VerifyBackupResult *backupResult = NULL;
 
-                        // Find the archiveId in the list - assert if not found since this should never happen
-                        String *archiveId = strLstGet(filePathLst, 0);
-                        unsigned int index = lstFindIdx(jobData.archiveIdResultList, &archiveId);
-                        ASSERT(index != LIST_NOT_FOUND);
+                        // Get archiveId result data
+                        if (strEq(fileType, STORAGE_REPO_ARCHIVE_STR))
+                        {
+                            // Find the archiveId in the list - assert if not found since this should never happen
+                            unsigned int index = lstFindIdx(jobData.archiveIdResultList, &resultId);
```

Comment:
As you noted in your email, this can be reduced to:
```
archiveIdResult = lstFind(jobData.archiveIdResultList, &resultId);
ASSERT(archiveIdResult != LIST_NOT_FOUND);
```